lineman/app/components/thread_preview/thread_preview.coffee

```diff
@@ -3,8 +3,17 @@ angular.module('loomioApp').directive 'threadPreview', ->
   restrict: 'E'
   templateUrl: 'generated/components/thread_preview/thread_preview.html'
   replace: true
-  controller: ($scope, Records, CurrentUser, LmoUrlService) ->
+  controller: ($scope, Records, CurrentUser, LmoUrlService, FlashService) ->
     $scope.lastVoteByCurrentUser = (thread) ->
       thread.activeProposal().lastVoteByUser(CurrentUser)
 
+    $scope.changeVolume = (volume) ->
+      $scope.previousVolume = $scope.thread.volume
+      $scope.thread.changeVolume(volume).then ->
```

Comment:
We've had a few conversations in the office about how we want less critical user interactions to return immediately rather than wait for the server response. Eg: Mark as read and mute/change volume. Doing this encourages users to do stuff quickly and they feel like the app is nice and fast.

Are you comfortable with updating thread.changeVolume to update the record locally before firing the network request? The flash would still depend on the request but the other parts of the UI (ie: the thread list) would update immediately.